pkgs/by-name/we/webdriverio/package.nix

```diff
@@ -0,0 +1,63 @@
+{ fetchFromGitHub
+, buildNpmPackage
+, lib
+, nodePackages
+, yq-go
+, makeWrapper
+, nodejs
+, fetchpatch
+, withChromium ? true, chromium
+, withChromedriver ? true, chromedriver
+, withFirefox ? true, firefox
+, withGeckodriver ? true, geckodriver
+}: buildNpmPackage rec {
+  pname = "webdriverio";
+  version = "8.32.2";
+
+  src = fetchFromGitHub {
+    owner = "webdriverio";
+    repo = "webdriverio";
+    rev = "v${version}";
+    hash = "sha256-hDHIyDrnqG28f2npZy4AIhkZ89h+XVYNiYtmDqeTpdI=";
+  };
+
+  patches = [
+    (fetchpatch {
+      name = "Regenerate-lockfile.patch";
+      url = "https://github.com/webdriverio/webdriverio/commit/33a588f9addacd55a48452c04ecf6a1b775fbc76.patch";
+      hash = "sha256-cM9fcSizzLIH73WChS0toNb3nGhLKEwrS76kRa5ZqBc=";
+    })
+    (fetchpatch {
+      name = "Add-chromedriver-env.patch";
+      url = "https://github.com/webdriverio/webdriverio/pull/12339.patch";
+      hash = "sha256-fsuR5JvgfTTtIxNR9S5XfjvAEEtNEof1wgj1HH5COa0=";
+    })
+  ];
+
+  npmDepsHash = "sha256-KOC9woFDCh+nTLBzuNQQ6fBwqDV5q3S/vcpP+8BGXlw=";
+  npmFlags = [ "--legacy-peer-deps" "--ignore-scripts" ];
+
+  nativeBuildInputs = [ yq-go nodePackages.rimraf makeWrapper ];
+
+  prePatch = ''
+    yq -i -o json '. + {"version":"${version}"}' package.json
+  '';
+
+  postInstall = ''
+    makeWrapper ${nodejs}/bin/node $out/bin/wdio \
+      --add-flags $out/lib/node_modules/webdriverio-monorepo/node_modules/@wdio/cli/bin/wdio.js \
+      ${lib.optionalString withFirefox "--prefix PATH : ${firefox}/bin"} \
+      ${lib.optionalString withChromium "--prefix PATH : ${chromium}/bin"} \
```

Comment:
What's the point of those options if it just takes the browser from PATH?

Reply:
good point actually